go/runtime/bundle/manager.go

```diff
@@ -212,9 +237,59 @@ func (m *Manager) run(ctx context.Context) {
 	}
 }
 
-// Add adds bundle from the given path.
-func (m *Manager) Add(path string) error {
-	manifest, err := m.explodeBundle(path)
+func dynamicManifestRewriter(labels map[string]string) ManifestRewriterFunc {
+	// When no labels are specified, do not perform any rewriting.
+	if len(labels) == 0 {
+		return nil
+	}
+
+	return func(manifest *Manifest) {
+		manifest.Name = ""
+		manifest.Version = version.Version{}
+
+		// Rewrite all components.
+		for _, comp := range manifest.Components {
+			h := tuplehash.New256(32, []byte("oasis-core/bundle: dynamic component ID"))
+			// Ensure components are isolated by origin.
+			_, _ = h.Write([]byte(labels[LabelOrigin]))
+
+			instanceID := labels[LabelInstanceID]
+			if instanceID == "" {
+				// When instance ID is not available, use the name of the component.
+				instanceID = comp.Name
+			} else {
+				// When instance ID is available, reset the component version.
+				comp.Version = version.Version{}
+			}
+			_, _ = h.Write([]byte(instanceID))
+
+			// Rewrite component name.
+			comp.Name = hex.EncodeToString(h.Sum(nil))
```

Comment:
Wouldn't it be better to follow git/docker convention by adding hash to the end of the name, e.g. `ROFL-MyAgent:401359af`.

Reply:
I want to avoid having user-controlled data in these components. For status, we can show labels and those can include more descriptive names if needed.